utility/src/main/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsLogScheduler.kt

```diff
@@ -0,0 +1,50 @@
+package org.oppia.android.util.logging.performancemetrics
+
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.PeriodicWorkRequest
+import androidx.work.WorkManager
+import org.oppia.android.util.logging.MetricLogScheduler
+import javax.inject.Inject
+
+private const val OPPIA_PERIODIC_METRIC_WORK = "OPPIA_PERIODIC_METRIC_WORK"
+private const val OPPIA_STORAGE_USAGE_WORK = "OPPIA_STORAGE_USAGE_WORK"
+private const val OPPIA_MEMORY_USAGE_WORK = "OPPIA_MEMORY_USAGE_WORK"
+
+/**
+ * Enqueues work requests for generating metric log reports for gaining an insight regarding into
+ * the performance of the application.
+ */
+class PerformanceMetricsLogScheduler @Inject constructor() : MetricLogScheduler {
+  override fun enqueueWorkRequestForPeriodicMetrics(
+    workManager: WorkManager,
+    workRequest: PeriodicWorkRequest
+  ) {
+    workManager.enqueueUniquePeriodicWork(
+      OPPIA_PERIODIC_METRIC_WORK,
+      ExistingPeriodicWorkPolicy.KEEP,
+      workRequest
+    )
+  }
+
+  override fun enqueueWorkRequestForStorageUsage(
+    workManager: WorkManager,
+    workRequest: PeriodicWorkRequest
+  ) {
+    workManager.enqueueUniquePeriodicWork(
+      OPPIA_STORAGE_USAGE_WORK,
+      ExistingPeriodicWorkPolicy.KEEP,
+      workRequest
+    )
+  }
+
+  override fun enqueueWorkRequestForMemoryUsage(
+    workManager: WorkManager,
+    workRequest: PeriodicWorkRequest
+  ) {
+    workManager.enqueueUniquePeriodicWork(
+      OPPIA_MEMORY_USAGE_WORK,
```

Comment:
Throughout this PR we are distinguishing between periodic and memory metrics--why? Aren't memory metrics also periodic?

Similarly, aren't some status like APK size and cold launch latency not periodic metrics?

Reply:
As decided in our initial discussions we had divided all these metrics into three main categories: 
- Periodic Metrics (cpu and network usage)
- App Startup Metrics (apk size, storage usage, startup latency)
- UI Specific Metrics (memory)

You're correct that memory metrics are also periodic but according to the TDD, they're tied to the UI and hence fall in a different category. 

Does that help?

Reply:
Hmm. It does (insomuch as I remember that discussion), but in terms of how we name these, I wonder if there's a better way to distinguish between the two types of periodic metrics (to avoid the potential misinterpretation that the UI metrics are *not* periodic). Perhaps: periodic_background_metrics and periodic_ui_metrics?

Reply:
done.